lib/eslint/eslint.js

```diff
@@ -284,6 +284,23 @@ function isFileTS(filePath) {
     return fileExtension.endsWith("ts");
 }
 
+/**
+ * Check if the current runtime supports native TypeScript.
+ * In those cases, we can use `import()` to load TypeScript files.
+ * @returns {boolean} `true` if the runtime supports TypeScript, `false` if it doesn't.
+ */
+function isRuntimeSupportsTS() {
+
+    /**
+     * We only do cheap early checks here to avoid requiring users to install `importx` in those cases,
+     * full checks are done by `importx` when it's used.
+     *
+     * It's known the Bun and Deno support TypeScript natively.
+     */
+    // eslint-disable-next-line no-undef -- Bun and Deno are global objects in their environments
+    return typeof Bun !== "undefined" || typeof Deno !== "undefined";
```

Comment:
We can avoid disabling rules by checking `globalThis`.
```suggestion
    return globalThis.Bun || globalThis.Deno;
```

Reply:
My worry was things might not always be on `global this.` (e.g. `require` and `__dirname` etc are actually on the local module wrapper). But yeah for `Bun` and `Deno` I guess they should be on global?